pennylane/variable.py

```diff
@@ -137,10 +140,11 @@ def val(self):
         # pylint: disable=unsubscriptable-object
         if self.name is None:
             # The variable is a placeholder for a positional argument
-            return self.free_param_values[self.idx] * self.mult
+            return Variable.free_param_values[self.idx] * self.mult
 
         # The variable is a placeholder for a keyword argument
-        if isinstance(self.kwarg_values[self.name], (Sequence, np.ndarray)):
-            return self.kwarg_values[self.name][self.idx] * self.mult
+        temp = Variable.kwarg_values[self.name]
+        if isinstance(temp, (Sequence, np.ndarray)):
+            return temp[self.idx] * self.mult
 
-        return self.kwarg_values[self.name] * self.mult
+        return temp * self.mult  # FIXME is this ever reached?
```

Comment:
This statement is reached if the keyword argument value is an int, float, bool, or str I suppose (in the latter two cases a multiplication makes no sense though). Is this never the case?

Reply:
Actually it can't be a str, since a string is a sequence.

Reply:
I think QNode wraps a scalar into a 1-sequence during the flattening. Only a single test reaches that final return statement.